lua/telescope/algos/linked_list.lua

```diff
@@ -2,6 +2,49 @@
 local LinkedList = {}
 LinkedList.__index = LinkedList
 
+---@brief [[
+--- A linked list consists of a collection of "nodes", which contain information about:
+--- - the identity of their predecessor, under the `prev` key
+--- - the identity of their successor, under the `next` key
+--- - some stored information associated to themselves, under the `item` key
+---
+--- So locally at each node the picture of the linked list looks like:
+--- <pre>
+---      ┌──────────┐ B.prev ┌──────────┐ C.prev ┌──────────┐
+--- ◄────│    A     │◄───────│    B     │◄───────│    C     │◄────
+---  ... │ ┌──────┐ │        │ ┌──────┐ │        │ ┌──────┐ │ ...
+--- ────►│ │A.item│ │───────►│ │B.item│ │───────►│ │C.item│ │────►
+---      │ └──────┘ │ A.next │ └──────┘ │ B.next │ └──────┘ │
+---      └──────────┘        └──────────┘        └──────────┘
+--- </pre>
+---
+--- There are two special nodes in the linked list, the `head` and the `tail`.
+--- The `head` is a node which has no predecessor, so `head.prev=nil`.
+--- The `tail` is a node which has no successor, so `head.next=nil`.
+---
+--- The number of nodes in the linked list is stored under the `size` key.
+---
+--- This implementation also allows a specific index to be kept track of.
+--- The index is called `track_at`, and the node at that index is called `_tracked_node`.
+--- The information stored in the tracked node i.e. `_tracked_node.item` is stored under the
+--- `tracked` key.
+---
+--- The global picture of a linked list looks like:
+--- <pre>
+--- ┌────────┐           ┌───────────────┐           ┌────────┐
+--- │ `head` │◄───   ◄───│ _tracked_node │◄───   ◄───│ `tail` │
+--- │ ┌────┐ │    ...    │   ┌───────┐   │    ...    │ ┌────┐ │
+--- │ │item│ │───►   ───►│   │tracked│   │───►   ───►│ │item│ │
+--- │ └────┘ │           │   └───────┘   │           │ └────┘ │
+--- └────────┘           └───────────────┘           └────────┘
+--- </pre>
+---@brief ]]
+
+
+--- Create a new linked list
+---@param opts table
+---   @key track_at number: the index of the node to track
```

Comment:
`@key` is currently not a thing, but that also doesn't matter because right now we are not exporting documentation for that file

Reply:
That's fine, was just matching what had been done elsewhere